src/west/manifest.py

```diff
@@ -124,25 +132,39 @@ def __init__(self, source_file=None, source_data=None):
 
         # Set up the public attributes documented above, as well as
         # any internal attributes needed to implement the public API.
-        self._load(self._data['manifest'])
+        self._load(self._data, sections)
 
     def get_remote(self, name):
         '''Get a manifest Remote, given its name.'''
         return self._remotes_dict[name]
 
-    def _malformed(self, complaint):
+    def _malformed(self, complaint, section='manifest'):
         context = (' file {} '.format(self.path) if self.path
                    else ' data:\n{}\n'.format(self._data))
         raise MalformedManifest('Malformed manifest{}(schema: {}):\n{}'
-                                .format(context, _SCHEMA_PATH, complaint))
+                                .format(context, _SCHEMA_PATH[section], complaint))
+
+    def _load(self, data, sections):
+
+        if 'west' in sections:
+            # Parse the west section first, if available
+            west = data.get('west', dict())
```

Comment:
Can create an empty dict with `{}`.

Reply:
I copied this from your code :)

Reply:
Hmm... don't remember adding that at least. :)